File: assn04/primes.pl
# This is my first Perl program.
# Prabhat Bhootra

# Function to test whether a number n is prime
sub isPrime
{
	$n = $_[0];
	if ($n <= 3) {
		return ($n > 1);	
	} elsif ($n % 2 == 0 || $n % 3 == 0) {
		return undef;
	} else {
	        $i = 5;
		while ($i*$i <= $n){
			if ($n % $i == 0 || $n % ($i + 2) == 0){
				return undef;
			}
			$i = $i + 6;
		}
	}
	return 1;
}

# Driver part of program to print all primes less than 100
for  ($y = 1; $y <= 100; $y++) {
	if (isPrime($y)) {
		print "$y\n";
	}
}

File: assn04/primes.php
<?php
/* This is my first PHP program. */
/* Prabhat Bhootra */

/* Function to check whether a number n is prime */
function isPrime($n) {
	if ($n <= 3){
		return ($n > 1);
	} elseif (($n % 2) == 0 || ($n % 3) == 0) {
		return false;
	} else {
		$i = 5;
		while ($i*$i <= $n) {
			if (($n % $i) == 0 || ($n % ($i + 2)) == 0) {
				return false;
			}
			$i = $i + 6;
		}
		return true;
	}
}

/* Driver part of program to print all primes less than 100 */
foreach (range(1, 100) as $x) {
	if (isPrime($x)) {
		echo "$x\n";
	}
}
?>
